src/robot_upstart/job.py

```diff
@@ -141,7 +141,7 @@ def add(self, package=None, filename=None, glob=None):
             for path in search_paths:
                 self.files.extend(glob_files(os.path.join(path, glob)))
 
-    def install(self, root="/", sudo="/usr/bin/sudo", Provider=providers.Upstart):
+    def install(self, root="/", sudo="/usr/bin/sudo", Provider=providers.detect_provider()):
```

Comment:
good practice in python not to call functions for default args, but specify None instead and do

```
if not Provider:
  Provider = detect_provider()
```

See http://www.informit.com/articles/article.aspx?p=2314818 item 20